pkg/providers/nutanix/validator.go

```diff
@@ -264,6 +299,34 @@ func (v *Validator) validateImageConfig(ctx context.Context, client Client, iden
 	return nil
 }
 
+func (v *Validator) validateTemplateMatchesKubernetesVersion(ctx context.Context, identifier anywherev1.NutanixResourceIdentifier, client Client, kubernetesVersionName string) error {
+	var templateName string
+	if identifier.Type == anywherev1.NutanixIdentifierUUID {
+		imageUUID := *identifier.UUID
+		imageDetails, err := client.GetImage(ctx, imageUUID)
+		if err != nil {
+			return fmt.Errorf("failed to find image with uuid %s: %v", imageUUID, err)
+		}
+		templateName = *imageDetails.Spec.Name
```

Comment:
`imageDetails.Spec` and `imageDetails.Spec.Name` are both pointers. Let's validate they aren't nil before we dereference them